### PR TITLE
SP-577: Correct IPAC Lead / SPHEREx POC logic

### PR DESCRIPTION
**Pull request by gpdf**

### CODE REVIEW

**Comment by gpdf** [comment]
My main concern here is that some categories of documents don't have "SPHEREx POC"s.  Previously that was enforced, in an incomplete way, by the code, but now it seems like it will just be by policy?

**Comment by jonathansick** [comment]
@gpdf Ok I'm done with my proposed changes. I think the problem with the TN series is that the "other authors" filter was filtering out the wrong role and the lead/POC property was incorrect. So what I'm proposing here is moving the properties for IPAC lead, SPHEREx POC, and "other authors" down to the base SpherexMetadata class so that they're always treated consistently.

This lines up with the the Jinja HTML template which is showing IPAC Lead, SPHEREx POC and other authors based on whether they're marked up in the document, irregardless of whether we expect a document to have those roles or not. This also matches the treatment in the LaTeX class code, because if someone added say a `\spherexlead` command to a document that didn't support it, the titling command would still add that spherexlead to the document.

Now, if we want to prevent a certain document series from having a SPHEREx POC, for example, we could add a Pydantic validator to its data model class that ensures no author is marked up with the SPHEREx Lead role. I think that would be easier to maintain and more transparent than what I had before.

I'll also mention that since the IPAC Lead and SPHEREx POC are marked up as roles on the "authors" (Contributor types), there's no change in the JSON serialized data. That's why we didn't need to version the properties after all because it's only the SPHEREx HTML template that was using those properties.

**Review by gpdf — COMMENTED** [review]
It looks good.  I can't actually approve because I opened the PR.  I think it makes sense if you merge  this, @jonathansick, because then you can restart things as needed as well.

**Review by jonathansick — APPROVED** [review]

File: src/spherexlander/themes/spherex/site/index.html.jinja [hunk 1]
```diff
@@ -50,26 +50,37 @@
   <div class="spherex-metadata-badges">
     {# Badge colors are taken from Brendan Crill's pipeline overview diagram. #}
     {% if metadata.pipeline_level %}
-      <img alt="Pipeline level {{ metadata.pipeline_level }}" src="https://img.shields.io/badge/Pipeline-{{ metadata.pipeline_level }}-{% if metadata.pipeline_level == "L1" %}88BFFE{% elif metadata.pipeline_level == "L2" %}1AC604{% elif metadata.pipeline_level == "L3" %}BEB1D0{% else %}FEBCFF{% endif %}">
+    <img alt="Pipeline level {{ metadata.pipeline_level }}"
+      src="https://img.shields.io/badge/Pipeline-{{ metadata.pipeline_level }}-{% if metadata.pipeline_level == " L1"
```
Comment (gpdf) [inline src/spherexlander/themes/spherex/site/index.html.jinja:54]: The extra space in `" L1"` has broken this logic.  See also below.

File: src/spherexlander/themes/spherex/site/index.html.jinja [hunk 2]
```diff
@@ -50,26 +50,37 @@
...
+      %}88BFFE{% elif metadata.pipeline_level=="L2" %}1AC604{% elif metadata.pipeline_level=="L3" %}BEB1D0{% else
+      %}FEBCFF{% endif %}">
     {% endif %}
     {% if metadata.diagram_index %}
-      <img alt="Diagram index {{ metadata.diagram_index }}" src="https://img.shields.io/badge/Diagram_Index-{{ metadata.diagram_index }}-{% if metadata.pipeline_level == "L1" %}88BFFE{% elif metadata.pipeline_level == "L2" %}1AC604{% elif metadata.pipeline_level == "L3" %}BEB1D0{% else %}FEBCFF{% endif %}">
+    <img alt="Diagram index {{ metadata.diagram_index }}"
+      src="https://img.shields.io/badge/Diagram_Index-{{ metadata.diagram_index }}-{% if metadata.pipeline_level == "
+      L1" %}88BFFE{% elif metadata.pipeline_level=="L2" %}1AC604{% elif metadata.pipeline_level=="L3" %}BEB1D0{% else
```
Comment (gpdf) [inline src/spherexlander/themes/spherex/site/index.html.jinja:61]: The line break inside "L1" appears to have broken this logic.

File: src/spherexlander/themes/spherex/site/index.html.jinja [hunk 3]
```diff
@@ -50,26 +50,37 @@
...
+      %}FEBCFF{% endif %}">
     {% endif %}
     {% if metadata.difficulty %}
-      <img alt="{{ metadata.difficulty }} difficulty" src="https://img.shields.io/badge/Difficulty-{{ metadata.difficulty }}-{% if metadata.difficulty == "High" %}red{% elif metadata.difficulty == "Medium" %}yellow{% elif metadata.difficulty == "Low" %}green{% else %}blue{% endif %}">
+    <img alt="{{ metadata.difficulty }} difficulty"
+      src="https://img.shields.io/badge/Difficulty-{{ metadata.difficulty }}-{% if metadata.difficulty == " High"
```
Comment (gpdf) [inline src/spherexlander/themes/spherex/site/index.html.jinja:66]: The extra space in " High" has broken this logic.

**Review by gpdf — COMMENTED** [review]
Hindsight is 20/20: this PR broke some of the logic in this template.  It appears that blanks were inadvertently added when some very long lines of if-elif-else logic were reflowed.

I will PR a fix.

**Pull request merged**